contracts/incentives/StakingRewards.sol

```diff
@@ -67,10 +68,7 @@ contract StakingRewards is Ownable, ReentrancyGuard {
         starttime = _starttime;
         DURATION = (_duration * 24 hours);
 
-        require(_initreward == IERC20(rewardToken).balanceOf(address(this)),
-                "StakingRewards: wrong reward amount supplied");
-
-        _notifyRewardAmount(_initreward);
+        rewardDistribution = msg.sender;
```

Comment:
In this case `msg.sender` would `FarmFactory` contract what seems to be not optimal, I guess we should pass `avatar` address to the contract so the `dao` has the `rewardDistribution` rights

Reply:
Good point - will rework this :)